### PR TITLE
Add a destroy command to wmcore-db-init

### PR DESCRIPTION
**Pull request by mmascher**
Hi,

I added an utility command to wmcore-db-init that allows to destroy a database so it can be recreated from scratch.

Can you have a look at it please?

### CODE REVIEW

File: bin/wmcore-db-init [hunk 1]
```diff
@@ -121,7 +128,18 @@ def create(config):
...
+
+def destroy(config):
+    params = {}
+
+    if hasattr(config.CoreDatabase, "tablespaceName"):
+        params["tablespace_table"] = config.CoreDatabase.tablespaceName
+    if hasattr(config.CoreDatabase, "indexspaceName"):
+        params["tablespace_index"] = config.CoreDatabase.indexspaceName
+
+    wmInit.clearDatabase(modules)
```
Comment (ticoann) [inline bin/wmcore-db-init]: params and modules are not really used in clearDabase method. You can just do wmInit.clearDatabase() which will delete all the data. (Unless you want to delete specific modules, in that case WMInit clearDatabase method need to be modified.

Comment (mmascher) [inline bin/wmcore-db-init]: I will remove them

**Comment by ticoann** [comment]
Hi Marco, if you want to add this in command line tool, could you add another safety check.
i.e. ask one more input to user with information, what is going to delete and type explicitly yes.
Such as following

```
Are you sure you want to delete all the wmagent database, it is not reversible. (yes, no)?
```

**Comment by mmascher** [comment]
Hi @ticoann , I have added an `askConfirmation` function to really check and make sure what the user is doing. FYI, this is gonna be used by CRAB3 devs to scratch their development database isntance.

File: bin/wmcore-db-init [hunk 2]
```diff
@@ -121,7 +128,18 @@ def create(config):
...
+    return
+
+def destroy(config):
+    params = {}
+
+    if hasattr(config.CoreDatabase, "tablespaceName"):
+        params["tablespace_table"] = config.CoreDatabase.tablespaceName
+    if hasattr(config.CoreDatabase, "indexspaceName"):
+        params["tablespace_index"] = config.CoreDatabase.indexspaceName
+
```
Comment (ticoann) [inline bin/wmcore-db-init]: Is setting 'params' needed? It doesn't seems to be used.

**Comment by ticoann** [comment]
Thanks, Marco. It looks good. I just made one comment on the destroy method.

File: bin/wmcore-db-init (hunk 2 above)
Comment (mmascher) [inline bin/wmcore-db-init]: Ops, I forgot to remove it! Done

**Comment by mmascher** [comment]
Thank you for your quick feedback!

**Comment by ticoann** [comment]
Thanks for the quick fix. :smile:

**Pull request merged**